libs/gi/db/src/Database/DataManagers/OptConfigDataManager.ts

```diff
@@ -307,3 +308,33 @@ export function handleArtSetExclusion(
   }
   return artSetExclusion
 }
+
+function mapFilterInplace<T>(xs: T[], f: (x: T) => T | undefined): T[] {
+  const a: T[] = []
+  let reuse = true
+  for (const x of xs) {
+    const y = f(x)
+    if (y) {
+      a.push(y)
+    }
+    reuse &&= x !== undefined && y === x
+  }
+  return reuse ? xs : a
+}
+
+function objectKeyMapInplace<K extends string, T>(
+  ks: readonly K[],
+  o: Record<K, T>,
+  f: (k: K) => T
+) {
+  const oKeys = Object.keys(o)
+  let reuse = oKeys.length == ks.length && ks.every((k, i) => k === oKeys[i])
```

Comment:
Can shortcut the logic?
```suggestion
  if(oKeys.length == ks.length && ks.every((k, i) => k === oKeys[i])) return o
```

Reply:
No, this doesn't validate the object's values